### ISSUE TITLE
Cypress errors on visit

### ISSUE THREAD

**Issue by Cubelaster** [cypress-io/cypress#3836]
### Current behavior:

https://monosnap.com/file/Lk5EhFPkym3M2NGqLIGPR0IxUAnGNJ

So, my problem is that whenever I try to run a test visiting my app for the first time (sometimes even more times), it just fails. 

It errors out on requests towards the backend and the error message is `net::ERR_UNEXPECTED`. 

It appears almost like it starts making requests the moment the HTML is loaded, and it doesn't see the `.js` files? Is that even possible? 

When i retry the test a couple of times, it works normal.

It seems to be an error connected to a browser. After it loads successfully, even opening new tabs and running tests in them, everything works fine. 

Even worse is trying to test production web. It seems Cypress can not get authorized? 
I constantly get `401 Unauthorized`.

We're running under Windows Authorization.

This seems like Cypress is out of sync? 

### Steps to reproduce:

This happens on EVERY solution i try to test.

In my company, there are a couple of solutions and all of them use a React + Web Api stacks. 
Front end is being built by webpack and server is running independently. Also, CORS is set up correctly. 

### Versions
Cypres: 3.2.0
Windows 10
Chrome 73.0.3683.86

**Comment by jennifer-shehane** [comment]
Hey @Cubelaster, I watched the video. Can you open your devtools while the tests are running to narrow down where the 'sort of undefined' error is coming from? This is being thrown from your application code somewhere. It may be an async thing because sometimes it is not happening.

**Comment by Cubelaster** [comment]
OK, so I've made another recording. This time it's as detailed as it gets. 
I might run the whole debug, but I don't think it matters much. 
The error really is coming from my code, BUT! 
This is an error that would happen if my server wasn't working. 
In such cases, I can't get a response from the back end and the front end just doesn't work. 
All of those requests are async and they are meant to be like that.
Because of this, it seems to me that this is a Cypress error. 
You can see in the example that after I let Cypress "sit still" while recording errors, and after refreshing it, it worked. 
It's like Cypress can't latch onto whatever it needs in time?
https://monosnap.com/file/NADeo3CH1aZmHehdwz9WAftUm0lMvG

**Comment by Cubelaster** [comment]
OK, there's another thing I noticed: 
I can't use cy.route. Or I simply don't know how. 
I'm trying to explicitly wait for request to end, but Cypress says that such request never happened?
I basically c/p-ed the request route to cy.route so it can't be wrong.
![image](https://user-images.githubusercontent.com/14044616/55138138-00cb0d80-5133-11e9-8aa9-9d38a174a107.png)

**Comment by Cubelaster** [comment]
Also, I can't send requests. Every time I get 401 - Unauthorized. 
Honestly, I think it's either CORS problem, or Cypress can't handle Win Auth.

**Comment by Cubelaster** [comment]
OK, so I've made progress. 
I implemented the hack mentioned here:
https://github.com/cypress-io/cypress/issues/95
and I kinda got going.
However, my initial problems are still there. I can't get initial load to work.

**Comment by Cubelaster** [comment]
OK, another status update: This still persists in Electron, but in Electron it's a bit different.
When testing in Chrome, every time I close Chrome window and start another test, the error repeats itself. Almost like Chrome loses "context". 
Electron has this problem only for first opening. After that, as long as Test Runner is open, Electron has the "context".

**Comment by jennifer-shehane** [comment]
Can you try listening for the uncaught exception, so we can at least get the error out of the way that is failing the test?

```js
Cypress.on('uncaught:exception', (err, runnable) => {
  debugger
  // returning false here prevents Cypress from
  // failing the test
  return false
})
```

**Comment by Cubelaster** [comment]
Will try that. Just give me some time. I'd really like to make use of Cypress cause it seems awesome but we can't run it if it fails our tests...

**Comment by jennifer-shehane** [comment]
Hey @Cubelaster, any updates on this?

**Comment by Cubelaster** [comment]
I will try this today.

**Comment by Cubelaster** [comment]
Unfortunately, not much use doing that:
![image](https://user-images.githubusercontent.com/14044616/56346227-f647e500-61c1-11e9-93ff-9469948d4fe3.png)
It just caught the same error it usually does... 
And, again, this error does come from my code, but in real application, this NEVER happens. Actually, this would happen if my back end wasn't working. Which is NOT the case here. And, again, after letting it sit for a couple of seconds, it connects just fine and runs normally...

**Comment by jennifer-shehane** [comment]
Have you tried going through the stack trace of this error to see the source of where the `'sort' of undefined` is being thrown?

Do you happen to be using [NTLM auth](https://github.com/cypress-io/cypress/issues/850)?

**Comment by Cubelaster** [comment]
I don't use NTLM, but I use WinAuth. For me, this is the source of my problems. It doesn't get verified by WinAuth ON TIME.
Can I do an explicit wait or something on visit? Or retry or something?

**Comment by jennifer-shehane** [comment]
@Cubelaster We have been playing with this idea of trying cy.visit(). @flotwig Would this use case retry within the issue you're working on? https://github.com/cypress-io/cypress/issues/4013

There's also a plugin to retry failed tests here: https://github.com/Bkucera/cypress-plugin-retries

**Comment by flotwig** [comment]
@jennifer-shehane I don't think #4013 will fix this issue - #4013 specifically only adds retries in the event of a network error (currently, any one of `['ECONNREFUSED', 'ECONNRESET', 'EPIPE']`), which it doesn't seem like this falls in to.

This error looks related to [this Chromium issue](https://bugs.chromium.org/p/chromium/issues/detail?id=51404) about ERR_UNEXPECTED with WinAuth. Maybe we need to do something special with Cypress's network layer or the way we launch Chrome to allow this to work as expected.

**Comment by Cubelaster** [comment]
This indeed seems like the described Chromium issue. 
Also, I don't know if I mentioned, but Electron "mode" has much less problems.
